### DISCUSSION TITLE
Hey google, what is the temperature in the living room?

### DISCUSSION THREAD

**Discussion by jabss** [andrei-tatar/node-red-contrib-smartnora#129] [General]
Hi,

Maybe what I'm asking isn't implemented in google side yet, just wanted to confirm if there is any way of doing it...

I have in nodered smart-nora working with some zigbee outlets. I say to google home "Hey google, turn on the table light" and it works beautifully!

Now I have another prototype with some nodered nodes that fetch the living room temperature from an influxdb database. The readings are reported periodically beforehand by other zigbee sensors and they get stored in the database. This prototype is basically a "inject" node, an "Influxdb in" node and a "debug" node.  I'm able to get the last recorded temperature.

Now, as you would imagine, the next step would be: "Hey google, what is the temperature in the living room?" or at least to have a widget in google home nest with the temperatures in the different rooms.

Is this possible? How? What smart-nora node could be used for this?

Thanks,
Joao

**Comment by andrei-tatar** [comment]
Hi,
Yes, it's possible. You can either use the sensor node (if you only want to report the temperature) or the thermostat node if you also want to control it.

Unfortunately Google Home android app doesn't show the value for the sensor node. But it does show it for the thermostat. But voice works fine.

Here's for example how I use it:
![image](https://user-images.githubusercontent.com/6473183/145446412-ff938ea3-1860-4f60-aaa8-441b16526e0a.png)

**Reply by jabss** [reply]
Thanks, I think I'm getting closer.

I have setup the smart-nora node and it appears in google home nest. However, the information doesn't seem to be accepted by the smart-nora node

`"[Sensor 1] ignoring property msg.payload.0"`

The payload I'm producing is:

`object
_msgid: "9e30c0bdfe609449"
payload: array[1]
0: object
time: "2021-12-10T11:15:57.000Z"
Temp: 16.9
topic: "googlehome"`

So I guess I should change this format so it could be understood by the smart-nora node.
What would be an acceptable object format?

Thanks

**Reply by andrei-tatar** [reply]
@jabss it's described here: https://github.com/andrei-tatar/node-red-contrib-smartnora/tree/master/doc/nodes/sensor

A message like this should do the trick:
```
{
  "payload": {
    "temperature": 22
  }
}
```

**Comment by jabss** [comment]
I think I got it, thanks.

Last question: Do I need an inject node to periodically trigger the database fetch in order to update the NORA node?
I realized that without it, google home "don't find the sensor". I'm either missing some configuration or this would be required.

Thanks

![image](https://user-images.githubusercontent.com/7312250/145631245-03549bdc-fc03-4c2e-b8f1-bd4a27bb9a8b.png)

**Reply by andrei-tatar** [reply]
I'm not sure if the DB node you're using knows to send a message when the data changes. If it doesn't, you need to poll it periodically and send it to NORA.

I personally inject the message from MQTT into NORA. So each time the value changes, I just map and forward it (the trigger node is to mark the node as offline in Google Home if a message isn't sent for a period of time 2.2hours). 
![image](https://user-images.githubusercontent.com/6473183/145632587-a359144b-280d-49a0-8525-00013db47c92.png)

**Reply by nielsblobel** [reply]
Hey, can someone help me with my problem im getting only a string value from my sensor but I need to bring in the right format for the NORA Smart home sensor NODE in Node-Red. If someone have a config for me or an idea that can work for me pls hit me up with an Massage.

**Reply by andrei-tatar** [reply]
@nielsblobel can you detail what is the input message format you get?

**Reply by nielsblobel** [reply]
From my sensor or waht ?

**Reply by nielsblobel** [reply]
Looks something like this
 msg.payload string "5.654678"
This is from the debug window in node-red 
This massage is coming from my plc via OPC UA Node

**Reply by andrei-tatar** [reply]
@nielsblobel I would just use a simple function node to convert the payload to a number and move it the temperature property:
![image](https://user-images.githubusercontent.com/6473183/157885854-0157d498-8037-49ff-b8c1-a1c7b181aaea.png)

```
[{"id":"3e02cb30c0440409","type":"inject","z":"b9a14f5917f73660","name":"","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"23.45","payloadType":"str","x":210,"y":500,"wires":[["4caf8f98a0a74733"]]},{"id":"1186b937303a7083","type":"noraf-sensor","z":"b9a14f5917f73660","devicename":"Sensor","roomhint":"","name":"","temperature":true,"unit":"C","humidity":false,"passthru":false,"nora":"3302889fa9b7cb44","topic":"","filter":false,"sensorSupport":false,"sensorType":"AirQuality","sensorStates":[],"sensorStatesThatNotify":[],"sensorNumeric":true,"openCloseSupport":false,"openCloseDiscrete":false,"onOffSupport":false,"x":590,"y":500,"wires":[]},{"id":"4caf8f98a0a74733","type":"function","z":"b9a14f5917f73660","name":"convert to temperature","func":"return {\n    payload: {\n        // + is to convert to a number, you can also use parseFloat\n        temperature: +msg.payload,\n    },\n};\n","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":400,"y":500,"wires":[["1186b937303a7083"]]},{"id":"3302889fa9b7cb44","type":"noraf-config","name":"NORA - test","group":"test","twofactor":"off","twofactorpin":"","localexecution":true,"structure":"","storeStateInContext":true}]
```

**Reply by nielsblobel** [reply]
Okay thanks ,
It's now working but the last problem is that the input Signal is to long for Google home have you an idea for a Funktion to round to only 2 digits after the , ?

**Reply by andrei-tatar** [reply]
@nielsblobel NORA automatically rounds the numbers to 1 decimal place.

Otherwise you can just use `Math.round` with some multiplication/division: `Math.round(msg.payload * 100) / 100` - will round to 2 decimal places
